Could not initialize class com.mojang.blaze3d.systems.RenderSystem starting from 1.19

### Operating System

Linux

### Version of Prism Launcher

Prism Launcher 7.2

### Version of Qt

Qt 5.15.10

### Description of bug

Every version prior to 1.19 works fine, but starting from 1.19 up to 1.20.2 fails to start with the same error message.
I've already reinstalled the launcher, doesn't help.
But everything works fine in the official launcher.

```
Prism Launcher version: 7.2 (Gentoo)


Launched instance in online mode

authserver.mojang.com resolves to:
    [52.85.49.78, 52.85.49.102, 52.85.49.12, 52.85.49.33]

session.minecraft.net resolves to:
    [13.107.246.45, 13.107.213.45, 2620:1ec:46::45, 2620:1ec:bdf::45]

textures.minecraft.net resolves to:
    [13.107.246.45, 13.107.213.45, 2620:1ec:46::45, 2620:1ec:bdf::45]

api.mojang.com resolves to:
    [13.107.213.45, 13.107.246.45, 2620:1ec:bdf::45, 2620:1ec:46::45]


Minecraft folder is:
/usr/home/bit6tream/.local/share/PrismLauncher/instances/1.20.2/.minecraft


Java path is:
/opt/openjdk-bin-17.0.6_p10/bin/java


Java is version 17.0.6, using 64 (amd64) architecture, from Eclipse Adoptium.


AMD Ryzen 5 1600 Six-Core Processor
Advanced Micro Devices, Inc. [AMD/ATI] Lexa PRO [Radeon 540/540X/550/550X / RX 540X/550/550X] (rev c7)

Subsystem: Micro-Star International Co., Ltd. [MSI] Lexa PRO [Radeon 540/540X/550/550X / RX 540X/550/550X]

Kernel driver in use: amdgpu

OpenGL version string: 4.6 (Compatibility Profile) Mesa 23.1.8

Main Class:
  net.minecraft.client.main.Main

Native path:
  /usr/home/bit6tream/.local/share/PrismLauncher/instances/1.20.2/natives

Traits:
traits XR:Initial
traits FirstThreadOnMacOS

Libraries:
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/org/lwjgl/lwjgl-glfw-natives-linux/3.3.2/lwjgl-glfw-natives-linux-3.3.2.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/org/lwjgl/lwjgl-glfw/3.3.2/lwjgl-glfw-3.3.2.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/org/lwjgl/lwjgl-jemalloc-natives-linux/3.3.2/lwjgl-jemalloc-natives-linux-3.3.2.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/org/lwjgl/lwjgl-jemalloc/3.3.2/lwjgl-jemalloc-3.3.2.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/org/lwjgl/lwjgl-natives-linux/3.3.2/lwjgl-natives-linux-3.3.2.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/org/lwjgl/lwjgl-openal-natives-linux/3.3.2/lwjgl-openal-natives-linux-3.3.2.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/org/lwjgl/lwjgl-openal/3.3.2/lwjgl-openal-3.3.2.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/org/lwjgl/lwjgl-opengl-natives-linux/3.3.2/lwjgl-opengl-natives-linux-3.3.2.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/org/lwjgl/lwjgl-opengl/3.3.2/lwjgl-opengl-3.3.2.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/org/lwjgl/lwjgl-stb-natives-linux/3.3.2/lwjgl-stb-natives-linux-3.3.2.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/org/lwjgl/lwjgl-stb/3.3.2/lwjgl-stb-3.3.2.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/org/lwjgl/lwjgl-tinyfd-natives-linux/3.3.2/lwjgl-tinyfd-natives-linux-3.3.2.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/org/lwjgl/lwjgl-tinyfd/3.3.2/lwjgl-tinyfd-3.3.2.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/org/lwjgl/lwjgl/3.3.2/lwjgl-3.3.2.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/com/github/oshi/oshi-core/6.4.5/oshi-core-6.4.5.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/com/google/code/gson/gson/2.10.1/gson-2.10.1.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/com/google/guava/failureaccess/1.0.1/failureaccess-1.0.1.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/com/google/guava/guava/32.1.2-jre/guava-32.1.2-jre.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/com/ibm/icu/icu4j/73.2/icu4j-73.2.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/com/mojang/authlib/5.0.47/authlib-5.0.47.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/com/mojang/blocklist/1.0.10/blocklist-1.0.10.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/com/mojang/brigadier/1.1.8/brigadier-1.1.8.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/com/mojang/datafixerupper/6.0.8/datafixerupper-6.0.8.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/com/mojang/logging/1.1.1/logging-1.1.1.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/com/mojang/patchy/2.2.10/patchy-2.2.10.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/com/mojang/text2speech/1.17.9/text2speech-1.17.9.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/commons-codec/commons-codec/1.16.0/commons-codec-1.16.0.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/commons-io/commons-io/2.13.0/commons-io-2.13.0.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/commons-logging/commons-logging/1.2/commons-logging-1.2.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/io/netty/netty-buffer/4.1.97.Final/netty-buffer-4.1.97.Final.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/io/netty/netty-codec/4.1.97.Final/netty-codec-4.1.97.Final.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/io/netty/netty-common/4.1.97.Final/netty-common-4.1.97.Final.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/io/netty/netty-handler/4.1.97.Final/netty-handler-4.1.97.Final.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/io/netty/netty-resolver/4.1.97.Final/netty-resolver-4.1.97.Final.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/io/netty/netty-transport-classes-epoll/4.1.97.Final/netty-transport-classes-epoll-4.1.97.Final.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/io/netty/netty-transport-native-epoll/4.1.97.Final/netty-transport-native-epoll-4.1.97.Final-linux-aarch_64.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/io/netty/netty-transport-native-epoll/4.1.97.Final/netty-transport-native-epoll-4.1.97.Final-linux-x86_64.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/io/netty/netty-transport-native-unix-common/4.1.97.Final/netty-transport-native-unix-common-4.1.97.Final.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/io/netty/netty-transport/4.1.97.Final/netty-transport-4.1.97.Final.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/it/unimi/dsi/fastutil/8.5.12/fastutil-8.5.12.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/net/java/dev/jna/jna-platform/5.13.0/jna-platform-5.13.0.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/net/java/dev/jna/jna/5.13.0/jna-5.13.0.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/net/sf/jopt-simple/jopt-simple/5.0.4/jopt-simple-5.0.4.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/org/apache/commons/commons-compress/1.22/commons-compress-1.22.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/org/apache/commons/commons-lang3/3.13.0/commons-lang3-3.13.0.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/org/apache/httpcomponents/httpclient/4.5.13/httpclient-4.5.13.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/org/apache/httpcomponents/httpcore/4.4.16/httpcore-4.4.16.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/org/apache/logging/log4j/log4j-api/2.19.0/log4j-api-2.19.0.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/org/apache/logging/log4j/log4j-core/2.19.0/log4j-core-2.19.0.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/org/apache/logging/log4j/log4j-slf4j2-impl/2.19.0/log4j-slf4j2-impl-2.19.0.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/org/joml/joml/1.10.5/joml-1.10.5.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/org/slf4j/slf4j-api/2.0.7/slf4j-api-2.0.7.jar
  /usr/home/bit6tream/.local/share/PrismLauncher/libraries/com/mojang/minecraft/1.20.2/minecraft-1.20.2-client.jar

Native libraries:

Params:
  --username  --version 1.20.2 --gameDir /usr/home/bit6tream/.local/share/PrismLauncher/instances/1.20.2/.minecraft --assetsDir /usr/home/bit6tream/.local/share/PrismLauncher/assets --assetIndex 8 --uuid  --accessToken  --userType  --versionType release

Window size: 854 x 480

Launcher: standard

Java Arguments:
[-Xms512m, -Xmx4096m, -Duser.language=en]


Minecraft process ID: 23529


[20:35:03] [Datafixer Bootstrap/INFO]: 192 Datafixer optimizations took 172 milliseconds
Exception caught from launcher
java.lang.NoClassDefFoundError: Could not initialize class com.mojang.blaze3d.systems.RenderSystem
	at ab.a(SourceFile:66)
	at eqv.a(SourceFile:2500)
	at eqv.a(SourceFile:2494)
	at net.minecraft.client.main.Main.main(SourceFile:223)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:88)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:126)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:71)
Exiting with ERROR
Process exited with code 2.
Clipboard copy at: 06 Oct 2023 20:36:24 +0300
```

### Steps to reproduce

* Start minecraft 1.19 - 1.20.2

### Suspected cause

No idea, 1.18.2 works just fine with the same setup.

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

Ok, so I updated the launcher and this time the java exceptions are a little different

```
Prism Launcher version: 8.0 (official)


Launched instance in online mode

authserver.mojang.com resolves to:
    [13.33.243.103, 13.33.243.48, 13.33.243.49, 13.33.243.122]

session.minecraft.net resolves to:
    [13.107.213.45, 13.107.246.45, 2620:1ec:bdf::45, 2620:1ec:46::45]

textures.minecraft.net resolves to:
    [13.107.246.45, 13.107.213.45, 2620:1ec:bdf::45, 2620:1ec:46::45]

api.mojang.com resolves to:
    [13.107.213.45, 13.107.246.45, 2620:1ec:bdf::53, 2620:1ec:46::53]


Minecraft folder is:
/usr/home/bit6tream/opt/prism/instances/1.20.2/.minecraft


Java path is:
/opt/openjdk-bin-17.0.8.1_p1/bin/java


Checking Java version...
Java is version 17.0.8.1, using 64 (amd64) architecture, from Eclipse Adoptium.


AMD Ryzen 5 1600 Six-Core Processor
Advanced Micro Devices, Inc. [AMD/ATI] Lexa PRO [Radeon 540/540X/550/550X / RX 540X/550/550X] (rev c7)

Subsystem: Micro-Star International Co., Ltd. [MSI] Lexa PRO [Radeon 540/540X/550/550X / RX 540X/550/550X]

Kernel driver in use: amdgpu

OpenGL version string: 4.6 (Compatibility Profile) Mesa 23.1.8

Main Class:
  net.minecraft.client.main.Main

Native path:
  /usr/home/bit6tream/opt/prism/instances/1.20.2/natives

Traits:
traits FirstThreadOnMacOS
traits XR:Initial

Libraries:
  /usr/home/bit6tream/opt/prism/libraries/org/lwjgl/lwjgl-glfw-natives-linux/3.3.2/lwjgl-glfw-natives-linux-3.3.2.jar
  /usr/home/bit6tream/opt/prism/libraries/org/lwjgl/lwjgl-glfw/3.3.2/lwjgl-glfw-3.3.2.jar
  /usr/home/bit6tream/opt/prism/libraries/org/lwjgl/lwjgl-jemalloc-natives-linux/3.3.2/lwjgl-jemalloc-natives-linux-3.3.2.jar
  /usr/home/bit6tream/opt/prism/libraries/org/lwjgl/lwjgl-jemalloc/3.3.2/lwjgl-jemalloc-3.3.2.jar
  /usr/home/bit6tream/opt/prism/libraries/org/lwjgl/lwjgl-natives-linux/3.3.2/lwjgl-natives-linux-3.3.2.jar
  /usr/home/bit6tream/opt/prism/libraries/org/lwjgl/lwjgl-openal-natives-linux/3.3.2/lwjgl-openal-natives-linux-3.3.2.jar
  /usr/home/bit6tream/opt/prism/libraries/org/lwjgl/lwjgl-openal/3.3.2/lwjgl-openal-3.3.2.jar
  /usr/home/bit6tream/opt/prism/libraries/org/lwjgl/lwjgl-opengl-natives-linux/3.3.2/lwjgl-opengl-natives-linux-3.3.2.jar
  /usr/home/bit6tream/opt/prism/libraries/org/lwjgl/lwjgl-opengl/3.3.2/lwjgl-opengl-3.3.2.jar
  /usr/home/bit6tream/opt/prism/libraries/org/lwjgl/lwjgl-stb-natives-linux/3.3.2/lwjgl-stb-natives-linux-3.3.2.jar
  /usr/home/bit6tream/opt/prism/libraries/org/lwjgl/lwjgl-stb/3.3.2/lwjgl-stb-3.3.2.jar
  /usr/home/bit6tream/opt/prism/libraries/org/lwjgl/lwjgl-tinyfd-natives-linux/3.3.2/lwjgl-tinyfd-natives-linux-3.3.2.jar
  /usr/home/bit6tream/opt/prism/libraries/org/lwjgl/lwjgl-tinyfd/3.3.2/lwjgl-tinyfd-3.3.2.jar
  /usr/home/bit6tream/opt/prism/libraries/org/lwjgl/lwjgl/3.3.2/lwjgl-3.3.2.jar
  /usr/home/bit6tream/opt/prism/libraries/com/github/oshi/oshi-core/6.4.5/oshi-core-6.4.5.jar
  /usr/home/bit6tream/opt/prism/libraries/com/google/code/gson/gson/2.10.1/gson-2.10.1.jar
  /usr/home/bit6tream/opt/prism/libraries/com/google/guava/failureaccess/1.0.1/failureaccess-1.0.1.jar
  /usr/home/bit6tream/opt/prism/libraries/com/google/guava/guava/32.1.2-jre/guava-32.1.2-jre.jar
  /usr/home/bit6tream/opt/prism/libraries/com/ibm/icu/icu4j/73.2/icu4j-73.2.jar
  /usr/home/bit6tream/opt/prism/libraries/com/mojang/authlib/5.0.47/authlib-5.0.47.jar
  /usr/home/bit6tream/opt/prism/libraries/com/mojang/blocklist/1.0.10/blocklist-1.0.10.jar
  /usr/home/bit6tream/opt/prism/libraries/com/mojang/brigadier/1.1.8/brigadier-1.1.8.jar
  /usr/home/bit6tream/opt/prism/libraries/com/mojang/datafixerupper/6.0.8/datafixerupper-6.0.8.jar
  /usr/home/bit6tream/opt/prism/libraries/com/mojang/logging/1.1.1/logging-1.1.1.jar
  /usr/home/bit6tream/opt/prism/libraries/com/mojang/patchy/2.2.10/patchy-2.2.10.jar
  /usr/home/bit6tream/opt/prism/libraries/com/mojang/text2speech/1.17.9/text2speech-1.17.9.jar
  /usr/home/bit6tream/opt/prism/libraries/commons-codec/commons-codec/1.16.0/commons-codec-1.16.0.jar
  /usr/home/bit6tream/opt/prism/libraries/commons-io/commons-io/2.13.0/commons-io-2.13.0.jar
  /usr/home/bit6tream/opt/prism/libraries/commons-logging/commons-logging/1.2/commons-logging-1.2.jar
  /usr/home/bit6tream/opt/prism/libraries/io/netty/netty-buffer/4.1.97.Final/netty-buffer-4.1.97.Final.jar
  /usr/home/bit6tream/opt/prism/libraries/io/netty/netty-codec/4.1.97.Final/netty-codec-4.1.97.Final.jar
  /usr/home/bit6tream/opt/prism/libraries/io/netty/netty-common/4.1.97.Final/netty-common-4.1.97.Final.jar
  /usr/home/bit6tream/opt/prism/libraries/io/netty/netty-handler/4.1.97.Final/netty-handler-4.1.97.Final.jar
  /usr/home/bit6tream/opt/prism/libraries/io/netty/netty-resolver/4.1.97.Final/netty-resolver-4.1.97.Final.jar
  /usr/home/bit6tream/opt/prism/libraries/io/netty/netty-transport-classes-epoll/4.1.97.Final/netty-transport-classes-epoll-4.1.97.Final.jar
  /usr/home/bit6tream/opt/prism/libraries/io/netty/netty-transport-native-epoll/4.1.97.Final/netty-transport-native-epoll-4.1.97.Final-linux-aarch_64.jar
  /usr/home/bit6tream/opt/prism/libraries/io/netty/netty-transport-native-epoll/4.1.97.Final/netty-transport-native-epoll-4.1.97.Final-linux-x86_64.jar
  /usr/home/bit6tream/opt/prism/libraries/io/netty/netty-transport-native-unix-common/4.1.97.Final/netty-transport-native-unix-common-4.1.97.Final.jar
  /usr/home/bit6tream/opt/prism/libraries/io/netty/netty-transport/4.1.97.Final/netty-transport-4.1.97.Final.jar
  /usr/home/bit6tream/opt/prism/libraries/it/unimi/dsi/fastutil/8.5.12/fastutil-8.5.12.jar
  /usr/home/bit6tream/opt/prism/libraries/net/java/dev/jna/jna-platform/5.13.0/jna-platform-5.13.0.jar
  /usr/home/bit6tream/opt/prism/libraries/net/java/dev/jna/jna/5.13.0/jna-5.13.0.jar
  /usr/home/bit6tream/opt/prism/libraries/net/sf/jopt-simple/jopt-simple/5.0.4/jopt-simple-5.0.4.jar
  /usr/home/bit6tream/opt/prism/libraries/org/apache/commons/commons-compress/1.22/commons-compress-1.22.jar
  /usr/home/bit6tream/opt/prism/libraries/org/apache/commons/commons-lang3/3.13.0/commons-lang3-3.13.0.jar
  /usr/home/bit6tream/opt/prism/libraries/org/apache/httpcomponents/httpclient/4.5.13/httpclient-4.5.13.jar
  /usr/home/bit6tream/opt/prism/libraries/org/apache/httpcomponents/httpcore/4.4.16/httpcore-4.4.16.jar
  /usr/home/bit6tream/opt/prism/libraries/org/apache/logging/log4j/log4j-api/2.19.0/log4j-api-2.19.0.jar
  /usr/home/bit6tream/opt/prism/libraries/org/apache/logging/log4j/log4j-core/2.19.0/log4j-core-2.19.0.jar
  /usr/home/bit6tream/opt/prism/libraries/org/apache/logging/log4j/log4j-slf4j2-impl/2.19.0/log4j-slf4j2-impl-2.19.0.jar
  /usr/home/bit6tream/opt/prism/libraries/org/joml/joml/1.10.5/joml-1.10.5.jar
  /usr/home/bit6tream/opt/prism/libraries/org/slf4j/slf4j-api/2.0.7/slf4j-api-2.0.7.jar
  /usr/home/bit6tream/opt/prism/libraries/com/mojang/minecraft/1.20.2/minecraft-1.20.2-client.jar

Native libraries:

Params:
  --username  --version 1.20.2 --gameDir /usr/home/bit6tream/opt/prism/instances/1.20.2/.minecraft --assetsDir /usr/home/bit6tream/opt/prism/assets --assetIndex 8 --uuid  --accessToken  --userType  --versionType release

Window size: 854 x 480

Launcher: standard

Java Arguments:
[-Xms512m, -Xmx4096m, -Duser.language=en]


Minecraft process ID: 23159


[00:13:04] [Datafixer Bootstrap/INFO]: 192 Datafixer optimizations took 148 milliseconds
Exception caught from launcher
java.lang.NoClassDefFoundError: Could not initialize class com.mojang.blaze3d.systems.RenderSystem
	at ab.a(SourceFile:66)
	at eqv.a(SourceFile:2500)
	at eqv.a(SourceFile:2494)
	at net.minecraft.client.main.Main.main(SourceFile:223)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:87)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:130)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70)
Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.UnsatisfiedLinkError: /tmp/lwjgl_bit6tream/3.3.2+13/x64/liblwjgl.so: /tmp/lwjgl_bit6tream/3.3.2+13/x64/liblwjgl.so: failed to map segment from shared object [in thread "Render thread"]
	at java.base/jdk.internal.loader.NativeLibraries.load(Native Method)
	at java.base/jdk.internal.loader.NativeLibraries$NativeLibraryImpl.open(NativeLibraries.java:388)
	at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:232)
	at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:174)
	at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2394)
	at java.base/java.lang.Runtime.load0(Runtime.java:755)
	at java.base/java.lang.System.load(System.java:1953)
	at org.lwjgl.system.Library.loadSystem(Library.java:189)
	at org.lwjgl.system.Library.loadSystemFromLibraryPath(Library.java:179)
	at org.lwjgl.system.Library.loadSystem(Library.java:132)
	at org.lwjgl.system.Library.loadSystem(Library.java:64)
	at org.lwjgl.system.Library.<clinit>(Library.java:52)
	at org.lwjgl.system.MemoryUtil.<clinit>(MemoryUtil.java:100)
	at ekn.<clinit>(SourceFile:8)
	at elk.<init>(SourceFile:48)
	at elr.<init>(SourceFile:19)
	at elr.<init>(SourceFile:23)
	at elr.<clinit>(SourceFile:11)
	at com.mojang.blaze3d.systems.RenderSystem.<clinit>(SourceFile:50)
	at net.minecraft.client.main.Main.main(SourceFile:211)
	... 3 more
Exiting with ERROR
Process exited with code 2.
Clipboard copy at: 22 Nov 2023 00:14:18 +0300
```

My `/tmp` has `noexec` option currently and I'm gonna try and remove it right now and see if minecraft starts working again.

--- a/launcher/Application.cpp
+++ b/launcher/Application.cpp
@@ -82,7 +82,6 @@
 
 #include <iostream>
 #include <mutex>
-#include <string_view>
 
 #include <QAccessible>
 #include <QCommandLineParser>
@@ -133,8 +132,13 @@
 #include "gamemode_client.h"
 #endif
 
-#if defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD) || defined(Q_OS_OPENBSD)
-#include <mntent.h>
+#if defined(Q_OS_LINUX)
+#include <sys/statvfs.h>
+#endif
+
+#if defined(Q_OS_FREEBSD) || defined(Q_OS_OPENBSD)
+#include <sys/mount.h>
+#include <sys/types.h>
 #endif
 
 #if defined(Q_OS_MAC)
@@ -993,35 +997,36 @@ Application::Application(int& argc, char** argv) : QApplication(argc, argv)
         }
     }
 
-#if defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD) || defined(Q_OS_OPENBSD)
-
     // notify user if /tmp is mounted with `noexec` (#1693)
     {
-        FILE* description = setmntent(MOUNTED, "r");
-        mntent* info = nullptr;
+        bool is_tmp_noexec = false;
 
-        while ((info = getmntent(description)) != nullptr) {
-            std::string_view directory = info->mnt_dir;
-            std::string_view options = info->mnt_opts;
+#if defined(Q_OS_LINUX)
 
-            if (directory == "/tmp" && options.rfind("noexec") != std::string_view::npos) {
-                auto infoMsg =
-                    tr("Your /tmp directory is currently mounted with the 'noexec' flag enabled.\n"
-                       "Some versions of Minecraft may not launch.\n");
-                auto msgBox = new QMessageBox(QMessageBox::Information, tr("Incompatible system configuration"), infoMsg, QMessageBox::Ok);
-                msgBox->setDefaultButton(QMessageBox::Ok);
-                msgBox->setAttribute(Qt::WA_DeleteOnClose);
-                msgBox->setMinimumWidth(460);
-                msgBox->adjustSize();
-                msgBox->open();
-                break;
-            }
-        }
+        struct statvfs tmp_stat;
+        statvfs("/tmp", &tmp_stat);
+        is_tmp_noexec = tmp_stat.f_flag & ST_NOEXEC;
 
-        endmntent(description);
-    }
+#elif defined(Q_OS_FREEBSD) || defined(Q_OS_OPENBSD)
+
+        struct statfs tmp_stat;
+        statfs("/tmp", &tmp_stat);
+        is_tmp_noexec = tmp_stat.f_flags & MNT_NOEXEC;
 
 #endif
+
+        if (is_tmp_noexec) {
+            auto infoMsg =
+                tr("Your /tmp directory is currently mounted with the 'noexec' flag enabled.\n"
+                   "Some versions of Minecraft may not launch.\n");
+            auto msgBox = new QMessageBox(QMessageBox::Information, tr("Incompatible system configuration"), infoMsg, QMessageBox::Ok);
+            msgBox->setDefaultButton(QMessageBox::Ok);
+            msgBox->setAttribute(Qt::WA_DeleteOnClose);
+            msgBox->setMinimumWidth(460);
+            msgBox->adjustSize();
+            msgBox->open();
+        }
+    }
 
     if (createSetupWizard()) {
         return;

--- a/launcher/Application.cpp
+++ b/launcher/Application.cpp
@@ -82,6 +82,7 @@
 
 #include <iostream>
 #include <mutex>
+#include <string_view>
 
 #include <QAccessible>
 #include <QCommandLineParser>
@@ -130,6 +131,10 @@
 #include <dlfcn.h>
 #include "MangoHud.h"
 #include "gamemode_client.h"
+#endif
+
+#if defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD) || defined(Q_OS_OPENBSD)
+#include <mntent.h>
 #endif
 
 #if defined(Q_OS_MAC)
@@ -987,6 +992,36 @@ Application::Application(int& argc, char** argv) : QApplication(argc, argv)
             FS::deletePath(update_success_marker.absoluteFilePath());
         }
     }
+
+#if defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD) || defined(Q_OS_OPENBSD)
+
+    // notify user if /tmp is mounted with `noexec` (#1693)
+    {
+        FILE* description = setmntent(MOUNTED, "r");
+        mntent* info = nullptr;
+
+        while ((info = getmntent(description)) != nullptr) {
+            std::string_view directory = info->mnt_dir;
+            std::string_view options = info->mnt_opts;
+
+            if (directory == "/tmp" && options.rfind("noexec") != std::string_view::npos) {
+                auto infoMsg =
+                    tr("Your /tmp directory is currently mounted with the 'noexec' flag enabled.\n"
+                       "Some versions of Minecraft may not launch.\n");
+                auto msgBox = new QMessageBox(QMessageBox::Information, tr("Incompatible system configuration"), infoMsg, QMessageBox::Ok);
+                msgBox->setDefaultButton(QMessageBox::Ok);
+                msgBox->setAttribute(Qt::WA_DeleteOnClose);
+                msgBox->setMinimumWidth(460);
+                msgBox->adjustSize();
+                msgBox->open();
+                break;
+            }
+        }
+
+        endmntent(description);
+    }
+
+#endif
 
     if (createSetupWizard()) {
         return;

